src/common/OwnAvatar.js

```diff
@@ -0,0 +1,46 @@
+/* @flow strict-local */
+import React, { PureComponent } from 'react';
+import { connect } from 'react-redux';
+
+import type { GlobalState, User } from '../types';
+import { getCurrentRealm, getOwnUser } from '../selectors';
+import ImageAvatar from './ImageAvatar';
+import { getFullUrl } from '../utils/url';
+import { getGravatarFromEmail } from '../utils/avatar';
+
+type Props = {|
+  user: User,
+  size: number,
+  realm: string,
+  shape: 'square' | 'rounded' | 'circle',
+|};
+
+/**
+ * Renders an image of the current user's avatar
+ *
+ * @prop [size] - Sets width and height in pixels.
+ * @prop [user] - Current presence for this user used to determine status.
+ * @prop [realm] - Current realm url, used if avatarUrl is relative.
+ * @prop [shape] - One of 'square', 'rounded', 'circle'.
+ */
+class OwnAvatar extends PureComponent<Props> {
```

Comment:
Huh. What's the motivation for adding this component? I.e., why not just say `<Avatar user={ownUser} ...`?

I'm not a fan of this duplication -- there are a bunch of details like "`shape` defaults to `rounded`", "`avatarUrl` from `avatar_url` or from gravatar like so", and others, that feel like they should only live in one place.

I appreciate that some of the wrinkles of `Avatar` are left out -- but that seems like it only lessens the duplication, and doesn't really provide a reason to do it.

Also I think this bit in the commit message is mistaken:
> does not render TextAvatar as it is only for group conversations

In fact, `GroupPmConversationItem` doesn't use this path -- it refers to `TextAvatar` directly.

Perhaps that `TextAvatar` path in `Avatar` never gets used at all? It might be something that applies only if we pass bogus data like an empty `email` etc., which we might no longer do and should probably stop doing if there's anywhere that we do. This is a bit of a tangled area of our code, so it'd be great to simplify it down.